system/jlib/jsocket.cpp

```diff
@@ -3179,6 +3207,40 @@ static bool decodeNumericIP(const char *text,unsigned *netaddr)
 
 static bool lookupHostAddress(const char *name,unsigned *netaddr)
 {
+    if (!name || !*name)
+        return false;
+
+    GetCachedHostName();
+
+    if (streq(name, "localhost"))
+    {
+        IpAddress ip;
+        ip.ipset(queryLocalIP());
+        ip.copyAddress(netaddr);
+        return true;
+    }
+    else if ( (streq(name, ".")) || (streq(name, myhostname.get())) )
+    {
+        IpAddress ip;
+        ip.ipset(queryHostIP());
+        ip.copyAddress(netaddr);
+        return true;
+    }
+    else if ( (!isdigit(name[0])) && (strchr(name, '.')) ) // fqdn name compare
+    {
+        char n1[256];
+        strcpy(n1, name);
+        int l1 = strcspn(n1, ".");
+        n1[l1] = '\0';
+        if (streq(n1, myhostname.get()))
+        {
+            IpAddress ip;
+            ip.ipset(queryHostIP());
+            ip.copyAddress(netaddr);
```

Comment:
We can debate if/how to handle comparing short and fqd names.

Reply:
can let DNS worry about that, if just resolve, then see if has resolved to local [isLocla()], then return queryHostIP() if local.

Reply:
This was the one thing I was unsure about - if we should specifically check for short to fqdn compare.  It can become tricky if both names are not in DNS.